app.js

```diff
@@ -0,0 +1,71 @@
+let books = [];
+
+const awesomeBooks = document.getElementById('awesomeBooks');
+
+function updateLocalstorage() {
+  localStorage.setItem('bookLibrary', JSON.stringify(books));
+}
+
+function remove(id) {
+  books = books.filter((book) => book.id !== id);
+  updateLocalstorage();
+}
+
+function removeDom(element) {
+  element.querySelectorAll('.btn').forEach((btn) => {
+    btn.addEventListener('click', (e) => {
+      const parent = e.target.parentNode;
+      remove(parent.id);
+      parent.remove();
+    });
+  });
+}
+
+function render(book) {
+  awesomeBooks.innerHTML += `
+      <li id="${book.id}">
+        <h3>${book.title}</h3>
+        <h4>${book.author}</h4>
+        <button class="btn">remove</button>
+        <hr>  
+      </li>
+    
+      `;
+  removeDom(awesomeBooks);
+}
+
+function add(book) {
+  render(book);
+  books.push(book);
```

Comment:
[OPTIONAL] The array push method can sometimes mutate the array, so you can consider using an array method like `concat` which does not mutate the array. :wink: